sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/connection/ShardingConnection.java

```diff
@@ -129,42 +134,64 @@ public Statement createStatement(final int resultSetType, final int resultSetCon
-            super.setAutoCommit(autoCommit);
-            return;
-        }
-        if (autoCommit && !shardingTransactionManager.isInTransaction() || !autoCommit && shardingTransactionManager.isInTransaction()) {
-            return;
-        }
-        if (autoCommit && shardingTransactionManager.isInTransaction()) {
-            shardingTransactionManager.commit();
-            return;
-        }
-        if (!autoCommit && !shardingTransactionManager.isInTransaction()) {
-            recordMethodInvocation(Connection.class, "setAutoCommit", new Class[]{boolean.class}, new Object[]{true});
-            closeCachedConnections();
-            shardingTransactionManager.begin();
-        }
-    }
-    
-    private void closeCachedConnections() throws SQLException {
-        getForceExecuteTemplate().execute(getCachedConnections().values(), new ForceExecuteCallback<Connection>() {
-            
-            @Override
-            public void execute(final Connection connection) throws SQLException {
-                connection.close();
-            }
-        });
-        getCachedConnections().clear();
-    }
+	public void setAutoCommit(final boolean autoCommit) throws SQLException {
+		if (this.originAutoCommit != null && this.originAutoCommit == autoCommit) {
+			return;
+		}
+
+		if (TransactionType.LOCAL == transactionType) {
+			super.setAutoCommit(autoCommit);
+			this.originAutoCommit = autoCommit;
+			return;
+		} else if (autoCommit) {
+			super.setAutoCommit(autoCommit);
+			this.originAutoCommit = autoCommit;
+			this.closeCachedConnections();
+			return;
+		}
+
+		this.transaction = this.shardingTransactionManager.suspend();
+		shardingTransactionManager.begin();
+		this.originAutoCommit = autoCommit;
+		this.closeCachedConnections();
+
+		// if (autoCommit && !shardingTransactionManager.isInTransaction() || !autoCommit &&
+		// shardingTransactionManager.isInTransaction()) {
+		// return;
+		// }
+		// if (autoCommit && shardingTransactionManager.isInTransaction()) {
+		// shardingTransactionManager.commit();
+		// return;
+		// }
+		// if (!autoCommit && !shardingTransactionManager.isInTransaction()) {
+		// recordMethodInvocation(Connection.class, "setAutoCommit", new Class[]{boolean.class}, new Object[]{true});
+		// closeCachedConnections();
+		// shardingTransactionManager.begin();
+		// }
+	}
+    
+	private void closeCachedConnections() throws SQLException {
+		getForceExecuteTemplate().execute(getCachedConnections().values(), new ForceExecuteCallback<Connection>() {
+
+			@Override
+			public void execute(final Connection connection) throws SQLException {
+				connection.close();
+			}
+		});
+		getCachedConnections().clear();
+	}
     
     @Override
     public void commit() throws SQLException {
         if (TransactionType.LOCAL == transactionType) {
             super.commit();
         } else {
             shardingTransactionManager.commit();
+            if (this.transaction != null) {
```

Comment:
both transaction and shardingTransacitonMananger appear on ShardingConnection, it is very Confused. transaction should managered by TransactionManager.